jina/serve/instrumentation/__init__.py

```diff
@@ -10,6 +10,26 @@
     from opentelemetry.metrics import Histogram
     from prometheus_client import Summary
 
+ENV_RESOURCE_ATTRIBUTES = [
+    'K8S_NAMESPACE_NAME',
+    'K8S_DEPLOYMENT_NAME',
+    'K8S_STATEFULSET_NAME',
+    'K8S_CLUSTER_NAME',
+    'K8S_NODE_NAME',
+    'K8S_POD_NAME',
+]
+
+
+def _get_resource_attributes(service_name: str):
+    from opentelemetry.semconv.resource import ResourceAttributes
+    import os
+
+    attributes = {ResourceAttributes.SERVICE_NAME: service_name}
+    for attribute in ENV_RESOURCE_ATTRIBUTES:
+        if attribute in os.environ:
+            attributes[ResourceAttributes.__dict__[attribute]] = os.environ[attribute]
```

Comment:
This line should be covered in the new test. Is codecov not happy because it is covered in the containers and not in the local jina execution?